### ISSUE TITLE
Connection leak to kube-dns causing large nf_conntrack table length

### ISSUE THREAD

**Issue by tanuck** [linkerd/linkerd2#3002]
### What is the issue?

After upgrading to `edge-19.6.3` an unknown interaction between `traefik` and `linkerd-proxy` is causing a steady leak of connections to `kube-dns` which results in large increases in `nf_conntrack_count`.
 
### How can it be reproduced?

Setup a GKE cluster with traefik as the ingress controller (our example traefik yaml linked below), create an Ingress object and annotate as per the instructions [here](https://linkerd.io/2/tasks/using-ingress/#traefik)
### Logs, error output, etc

Not sure, we have a "broken" cluster running still with 19.6.3 so can provide as needed.

#### `linkerd check` output

```text
$ linkerd check
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist

linkerd-existence
-----------------
√ control plane components ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ no invalid service profiles

linkerd-version
---------------
√ can determine the latest version
√ cli is up-to-date

control-plane-version
---------------------
√ control plane is up-to-date
√ control plane and cli versions match

Status check results are √
```

### Environment

- Kubernetes Version: 1.13.6-gke.13
- Cluster Environment: GKE
- Host OS: cos
- Linkerd version: edge-19.6.3

### Additional context
[Example Traefik setup](https://gist.github.com/tanuck/f86eeb7cb04d776cc0180da0da0255c0) - close as we could get to our setup

**Comment by tanuck** [comment]
I think we've confirmed that this is not just a traefik-linkerd issue, it would seem that all pods in the service mesh are leaking connections to `kube-dns` - however the rate is far more noticeable for the ingress controller.

**Comment by olix0r** [comment]
@tanuck When you say "connections", do you mean TCP connections? Or are we tracking UDP file descriptors? Or something else?

Is there any DNS configuration present in these pods?

It would surprise me if Linkerd is initiating TCP connections to the DNS server, but it's definitely possible...

**Comment by tanuck** [comment]
"Connections" probably not the optimum word. No the "leaking" entries within `nf_conntrack` seem to be made up of `ASSURED` UDP connection states. All src hosts are meshed pods and dst is kubedns pod.

**Comment by tanuck** [comment]
I think I've isolated the issue to the following commit in the proxy:
https://github.com/linkerd/linkerd2-proxy/commit/52eefa45c481cd68eff8f6785ea8eb8b47c0569b

![screenshot(1)](https://user-images.githubusercontent.com/289004/60598191-2b68da00-9da4-11e9-9cd8-3daded04efab.png)

**Comment by grampelberg** [comment]
That’s awesome, thank you for the help @tanuck !!

**Comment by tanuck** [comment]
Tried out `edge-19.7.1` today - the problem still persists.

**Comment by olix0r** [comment]
Sorry, @tanuck -- the fix was implemented after that edge release. We plan to do an edge-19.7.2 with the fix.

**Comment by tanuck** [comment]
You’ve got a fix already?

**Comment by olix0r** [comment]
@tanuck We just released edge-19.7.2 which at least fixes one such leak -- we had introduced a change to more eagerly tear down idle clients to free their resources. However, the DNS lookups for this client were not properly torn down, and so if the client was recreated later, we'd start polling for DNS updates again, doubling the number of active UDP sessions.  To fix this, we now properly teardown DNS lookups when clients are idled out.

```
:; curl -sL https://run.linkerd.io/install-edge |sh -
Downloading linkerd2-cli-edge-19.7.2-linux...
```

Please do test it and let us know if you see anything amiss!

**Comment by vic3lord** [comment]
Thanks @olix0r, hope the latest edge do solve this, I came here to update that we encountered the same issue and I have screenshot from today.

We downgraded the version from edge to latest stable and this is the effect on kube-dns
<img width="669" alt="image" src="https://user-images.githubusercontent.com/2883824/60835267-55b6ff00-a1cb-11e9-9bd6-b68f69673afa.png">

Hope this helps anyone else

**Comment by tanuck** [comment]
Great work @olix0r - we'll give `19.7.2` a try today!

**Comment by tanuck** [comment]
The issue seems resolved with `edge-19.7.2`

![Screenshot_2019-07-09 Conntrack - Grafana](https://user-images.githubusercontent.com/289004/60888931-c30c7380-a24f-11e9-8571-d03f295fb338.png)

**Comment by wmorgan** [comment]
Great!!!
